stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -354,85 +354,123 @@ function generic_vecnormp(x, p)
     end
 end
 
-vecnormMinusInf(x) = generic_vecnormMinusInf(x)
-vecnormInf(x) = generic_vecnormInf(x)
-vecnorm1(x) = generic_vecnorm1(x)
-vecnorm2(x) = generic_vecnorm2(x)
-vecnormp(x, p) = generic_vecnormp(x, p)
+normMinusInf(x) = generic_normMinusInf(x)
+normInf(x) = generic_normInf(x)
+norm1(x) = generic_norm1(x)
+norm2(x) = generic_norm2(x)
+normp(x, p) = generic_normp(x, p)
+
 
 """
-    vecnorm(A, p::Real=2)
+    norm(A, p::Real=2)
 
 For any iterable container `A` (including arrays of any dimension) of numbers (or any
 element type for which `norm` is defined), compute the `p`-norm (defaulting to `p=2`) as if
 `A` were a vector of the corresponding length.
 
-The `p`-norm is defined as:
+The `p`-norm is defined as
 ```math
 \\|A\\|_p = \\left( \\sum_{i=1}^n | a_i | ^p \\right)^{1/p}
 ```
-with ``a_i`` the entries of ``A`` and ``n`` its length.
+with ``a_i`` the entries of ``A``, ``| a_i |`` are their [`norm`](@ref)s, and
+``n`` the length of ``A``. Since the `p`-norm is computed using the [`norm`](@ref)s
+of the entries of `A`, the `p`-norm of a vector of vectors is not compatible with
+the interpretation of it as a block vector in general if `p != 2`.
 
 `p` can assume any numeric value (even though not all values produce a
-mathematically valid vector norm). In particular, `vecnorm(A, Inf)` returns the largest value
-in `abs(A)`, whereas `vecnorm(A, -Inf)` returns the smallest. If `A` is a matrix and `p=2`,
+mathematically valid vector norm). In particular, `norm(A, Inf)` returns the largest value
+in `abs.(A)`, whereas `norm(A, -Inf)` returns the smallest. If `A` is a matrix and `p=2`,
 then this is equivalent to the Frobenius norm.
 
+The second argument `p` is not necessarily a part of the interface for `norm`, i.e. a custom
+type may only implement `norm(A)` without second argument.
+
+Use [`opnorm`](@ref) to compute the operator norm of a matrix.
+
 # Examples
 ```jldoctest
-julia> vecnorm([1 2 3; 4 5 6; 7 8 9])
+julia> v = [3, -2, 6]
+3-element Array{Int64,1}:
+  3
+ -2
+  6
+
+julia> norm(v)
+7.0
+
+julia> norm(v, 1)
+11.0
+
+julia> norm(v, Inf)
+6.0
+
+julia> norm([1 2 3; 4 5 6; 7 8 9])
+16.881943016134134
+
+julia> norm([1 2 3 4 5 6 7 8 9])
 16.881943016134134
 
-julia> vecnorm([1 2 3 4 5 6 7 8 9])
+julia> norm([1 2 3 4 5 6 7 8 9]')
```

Comment:
Perhaps a vector of 1:9 would be more instructive alongside the preceding examples than the adjoint of a single-row matrix?